src/chocolatey/infrastructure.app/configuration/ChocolateyConfiguration.cs

```diff
@@ -384,6 +384,8 @@ public sealed class SourcesCommandConfiguration
         public string Username { get; set; }
         public string Password { get; set; }
         public int Priority { get; set; }
+        public string Certificate { get; set; }
+        public string CertificatePassword { get; set; }

```

Comment:
Are you saying sometimes you would authenticate to the ***same*** source with user name / password and sometimes you would authenticate with the certificate / password? Because I'm having a really hard time seeing that.